File: leetcode/1021_Remove_Outermost_Parentheses.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    string removeOuterParentheses(string s) {
        int open=0, close=0;
        string str="";
        int l=0;
        for(int i=0;i<s.size();i++){
            if(s[i]=='(') open++;
            else close++;
            if(open==close){
                str.insert(str.end(),s.begin()+l+1,s.begin()+i);
                l=i+1;
                open=close=0;
            }    
        }
        return str;
    }
};
int main(){
    Solution s;
    string str="(()())(())";
    cout<<s.removeOuterParentheses(str);
}

File: leetcode/1021_Remove_Outermost_Parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        int open=0;
        StringBuilder res=new StringBuilder();
        for(char c:s.toCharArray()){
            if(c=='('){
                if(open>0){
                    res.append('(');
                }
                open++;
            }
            else{
                open--;
                if(open>0)res.append(')');
            }
        }
        return res.toString();
    }
}